src/data/cards.js

```diff
@@ -0,0 +1,221 @@
+        }
+        for(let card of cardList){
+            card._id = card._id.toString();
+        }
+        return cardList;
+    },
+
+    /**
+     * Get a specific card from a board.
+     * @param {string} boardId The board's id.
+     * @param {string} cardId The card's id.
+     * @returns A card object
+     */
+    async getCardById(boardId, cardId){
+        if(!boardId || !error_handler.checkObjectId(boardId)){
+            throw new Error('boardId is not valid');
+        }
+        if(!cardId || !error_handler.checkObjectId(cardId)){
+            throw new Error('cardId is not valid');
+        }
+
+        const boardCollection = await boards();
+        const board = await boardCollection.findOne({_id: ObjectId(boardId)});
+        if(board === null){
+            throw new Error('There is not board with that id.');
+        }
+        let foundCard;
+        
+        for(let card of board.cards){
+            if(card._id.toString() === cardId){
+                foundCard = card;
+            }
+        }
+        if(!foundCard){
+            throw new Error('Card not found');
+        } 
+        foundCard._id = foundCard._id.toString();
+
+        return foundCard;
+    }, 
+
+    /**
+     * Add a new card to a board.
+     * @param {string} boardId The board's id.
+     * @param {string} listId The list's id.
+     * @param {string} cardName The name of the card.
+     * @returns The new card.
+     */
+    async addCard(boardId, listId, cardName){
+        if(!boardId || !error_handler.checkObjectId(boardId)){
+            throw new Error('boardId is not valid');
+        }
+        if(!listId || !error_handler.checkObjectId(listId)){
+            throw new Error('listId is not valid');
+        }
+        if(!cardName || !error_handler.checkNonEmptyString(cardName)){
+            throw new Error('cardName must not be empty');
+        }
+
+        let newBoardId = ObjectId(boardId);
+        const boardCollection = await boards();
+
+        //storyPoints and dueDate can be added via update, but are not created with a new card
+        const newCard = {
+            _id: new ObjectId(), 
+            cardName: cardName,
+            description: '',
+            labels: [],
+            comments: [],
+            assigned: [],
+            list: listId
+        };
+
+        const updateInfo = await boardCollection.updateOne({_id: newBoardId}, { $push: { cards: newCard}});
+
+        if (!updateInfo.matchedCount && !updateInfo.modifiedCount){
+            throw new Error('Could not add the card.');
+        }
+
+        const card = await getCardById(boardId, newCard._id)
+        return card;
+
+    },
+
+    /**
+     * Updates an existing card.
+     * @param {string} boardId The board's id.
+     * @param {string} cardId The card's id.
+     * @param {string} listId the list's id.
+     * @param {string} cardName The name of the card.
+     * @param {number} storyPoints The story point of the card.
```

Comment:
Everything may be passed in as a string because of how the post request would work, unless we use ajax.